File: JavaScript/110. Balanced Binary Tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isBalanced = function(root) {
    if(helper(root)!=-1){
        return true;
    }
    return false;
}
var helper = function(root){
    if(root==null){
        return 0;
    }
    var l=helper(root.left);
    if(l==-1){
        return -1;
    }
    var r=helper(root.right);
    if(r==-1){
        return -1;
    }
    if(Math.abs(l-r)>1){
        return -1;
    }
    return 1+Math.max(l,r);
};

File: C#/110. Balanced Binary Tree.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public bool IsBalanced(TreeNode root) {
        if(helper(root)!=-1){
            return true;
        }
        return false;
    }
    public int helper(TreeNode root){
        if(root==null){
            return 0;
        }
        int l=helper(root.left);
        if(l==-1){
            return -1;
        }
        int r=helper(root.right);
        if(r==-1){
            return -1;
        }
        if(Math.Abs(l-r)>1){
            return -1;
        }
        return 1+Math.Max(l,r);
    }
}
